lib/rules/prefer-arrow-callback.js

```diff
@@ -81,42 +101,62 @@ function getCallbackInfo(node) {
 //------------------------------------------------------------------------------
 
 module.exports = function(context) {
-    // {Array<boolean>}
-    // This stack stores flags that function scopes have one or more ThisExpressions or not.
+    // {Array<{this: boolean, super: boolean, meta: boolean}>}
+    // - this - A flag which shows there are one or more ThisExpression.
+    // - super - A flag which shows there are one or more Super.
+    // - meta - A flag which shows there are one or more MethProperty.
     var stack = [];
 
     /**
-     * Pushes new function scope with `false` flag.
-     * That `false` is meaning the scope does not have ThisExpression.
+     * Pushes new function scope with all `false` flags.
      * @returns {void}
      */
     function enterScope() {
-        stack.push(false);
+        stack.push({this: false, super: false, meta: false});
     }
 
     /**
      * Pops a function scope from the stack.
-     * @returns {boolean} `true` if the function scope has one or more ThisExpressions.
+     * @returns {{this: boolean, super: boolean, meta: boolean}} The information of the last scope.
      */
     function exitScope() {
         return stack.pop();
     }
 
     return {
+        // Reset internal state.
         Program: function() {
             stack = [];
         },
 
+        // If there are below, it cannot replace with arrow functions merely.
         ThisExpression: function() {
-            stack[stack.length - 1] = true;
+            var info = stack[stack.length - 1];
+            if (info) {
+                info.this = true;
+            }
+        },
+        Super: function() {
+            var info = stack[stack.length - 1];
+            if (info) {
+                info.super = true;
+            }
+        },
+        MetaProperty: function() {
```

Comment:
The former is not a good excuse. It will just need updating in the near future. The latter seems like it should just be considered a blocker for this issue. You can't do this right until that bug is fixed.